components/org.wso2.carbon.identity.api.server.idp/org.wso2.carbon.identity.api.server.idp.v1/src/main/java/org/wso2/carbon/identity/api/server/idp/v1/core/ServerIdpManagementService.java

```diff
@@ -2733,55 +2742,124 @@ private void processPatchRequest(List<Patch> patchRequest, IdentityProvider idpT
                             patchIdpProperties(idpToUpdate, Constants.IDP_ISSUER_NAME, value);
                             break;
                         case Constants.CERTIFICATE_JWKSURI_PATH:
-                            patchJWKSURI(idpToUpdate, Constants.JWKS_URI, value);
+                            patchIdpProperties(idpToUpdate, Constants.JWKS_URI, value);
                             break;
                         default:
                             throw handleException(Response.Status.BAD_REQUEST, Constants.ErrorMessage
                                     .ERROR_CODE_INVALID_INPUT, null);
                     }
                 }
-            } else if (operation == Patch.OperationEnum.ADD && path.matches(Constants.CERTIFICATE_PATH_REGEX) && path
-                    .split(Constants.PATH_SEPERATOR).length == 4) {
-                List<String> certificates = new ArrayList<>();
-                int index = Integer.parseInt(path.split(Constants.PATH_SEPERATOR)[3]);
-                if (index < 0) {
-                    throw handleException(Response.Status.BAD_REQUEST, Constants.ErrorMessage
-                            .ERROR_CODE_INVALID_INPUT, "Invalid index in 'path' attribute");
-                }
-                if (ArrayUtils.isNotEmpty(idpToUpdate.getCertificateInfoArray())) {
-                    if (index > idpToUpdate.getCertificateInfoArray().length) {
-                        throw handleException(Response.Status.BAD_REQUEST, Constants.ErrorMessage
-                                .ERROR_CODE_INVALID_INPUT, "Invalid index in 'path' attribute");
+            } else if (operation == Patch.OperationEnum.ADD) {
+                if (isCertificateUpdateRequest) {
+
+                    List<String> certificates = new ArrayList<>();
+                    int index = Integer.parseInt(path.split(Constants.PATH_SEPERATOR)[3]);
```

Comment:
Wouldn't this throw a NumberFormatException?

Reply:
Related to the dependent PR